backend/mbql/src/metabase/mbql/util.clj

```diff
@@ -453,10 +453,9 @@
     (field-clause->id-or-literal [:datetime-field [:field-id 100] ...]) ; -> 100
     (field-clause->id-or-literal [:field-id 100])                       ; -> 100
 
-  For expressions (or any other clauses) this returns the clause as-is, so as to facilitate the primary use case of
-  comparing Field clauses."
+  For expressions returns the expression name."
   [clause :- mbql.s/Field]
-  (second (unwrap-field-clause clause)))
+  (second (maybe-unwrap-field-clause clause)))
```

Comment:
This works currently, but `second` here implies an argument order that might not work with future additions to MBQL. I could see an argument, to forgo `maybe-unwrap-field-clause` in favour of `unwrap-field-clause` and repeat the dispatch logic.